### PR TITLE
Add Containerfile for amd/ROCM

### PR DESCRIPTION
**Pull request by ericcurtin**
Untested, might need some trimming, it's 10 G in size.

### CODE REVIEW

**Comment by ericcurtin** [comment]
This container image is 10G in size now... With some very aggressive optimization for size, it may be possible to get it down to 6G... But I don't think it will solve the issue long-term...

**Comment by p5** [comment]
In Universal Blue we have an action to cleanup unused dependencies before the build, which removes things like Android tools, browsers etc.  It frees up about 9GB, so maybe not enough for this use-case.

**Comment by cooktheryan** [comment]
is there any way we can multistep the build like this https://github.com/containers/fetchit/blob/main/Dockerfile

File: container-images/ramalama/latest-amd/Containerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM quay.io/ramalama/ramalama:latest
+
+RUN echo -e "[ROCm]\nname=ROCm\nbaseurl=https://repo.radeon.com/rocm/rhel9/6.2/main\nenabled=1\ngpgcheck=0\npriority=50" >> /etc/yum.repos.d/rocm.repo
+RUN echo -e "[amdgpu]\nname=amdgpu\nbaseurl=https://repo.radeon.com/amdgpu/6.2/rhel/9.4/main/x86_64\nenabled=1\ngpgcheck=0\npriority=50" >> /etc/yum.repos.d/amdgpu.repo
+
+RUN dnf install -y rocm-dev hipblas-devel rocblas-devel && \
+    dnf clean all && \
+    rm -rf /var/cache/*dnf*
+
+RUN git clone https://github.com/ggerganov/llama.cpp && \
```
Comment (rhatdan) [inline container-images/ramalama/latest-amd/Containerfile:10]: --depth 0

File: container-images/ramalama/latest-amd/Containerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+FROM quay.io/ramalama/ramalama:latest
+
+RUN echo -e "[ROCm]\nname=ROCm\nbaseurl=https://repo.radeon.com/rocm/rhel9/6.2/main\nenabled=1\ngpgcheck=0\npriority=50" >> /etc/yum.repos.d/rocm.repo
+RUN echo -e "[amdgpu]\nname=amdgpu\nbaseurl=https://repo.radeon.com/amdgpu/6.2/rhel/9.4/main/x86_64\nenabled=1\ngpgcheck=0\npriority=50" >> /etc/yum.repos.d/amdgpu.repo
+
+RUN dnf install -y rocm-dev hipblas-devel rocblas-devel && \
```
[the remark this answers is not in the document]
Comment (ericcurtin) [reply container-images/ramalama/latest-amd/Containerfile:6]: This is one thing I am not sure about...

Would need some digging, iterations, testing (most importantly testing of the various GPU types) to figure out what is a runtime dependency vs buildtime dependency...

I do kinda wish we up ubi9-based containers in quay.io that were agnostic to tools like ramalama and instructlab for the community like:

```
llama.cpp-cpu    vllm-cpu
llama.cpp-rocm   vllm-rocm
llama.cpp-cuda   vllm-cuda
llama.cpp-vulkan vllm-vulkan
```

outside of the ramalama project, by people with access to container image building infrastructure, a variety GPUs, etc.

Maybe some AI gurus want to use vllm and llama.cpp directly, which is fine!

These might be hard to maintain in ramalama repo, using commonly available github CI.

But we would happily use these as base images :)

[the remark this answers is not in the document]
Comment (ericcurtin) [reply container-images/ramalama/latest-amd/Containerfile:6]: Nvidia actually maintain ubi9 images for cuda which could act as a base image for those types:

nvcr.io/nvidia/cuda:12.6.1-cudnn-devel-ubi9
nvcr.io/nvidia/cuda:12.6.1-cudnn-runtime-ubi9
nvcr.io/nvidia/cuda:12.6.1-devel-ubi9
nvcr.io/nvidia/cuda:12.6.0-cudnn-devel-ubi9

available on nvcr.io and dockerhub.

[the remark this answers is not in the document]
Comment (rhatdan) [reply container-images/ramalama/latest-amd/Containerfile:6]: We could build a base image with two images sharing nvidia-lamacpp nvidia-vllm

[the remark this answers is not in the document]
Comment (ericcurtin) [reply container-images/ramalama/latest-amd/Containerfile:6]: It all comes down to size I guess... If combining nvidia-lamacpp and nvidia-vllm into a single image doesn't result in GBs of bloat for both... Would make sense

[the remark this answers is not in the document]
Comment (rhatdan) [reply container-images/ramalama/latest-amd/Containerfile:6]: Yup, depends on bloat, easier to have them both, but I am fine with them being separate.

**Pull request closed without merging**